feat: add namespace validation in namespace selectors (#2613)
kyma-project/telemetry-manager: docs/user/resources/05-metricpipeline.md

| **input.&#x200b;istio.&#x200b;diagnosticMetrics**  | object | DiagnosticMetrics configures collection of additional diagnostic metrics. The default is `false`. |
| **input.&#x200b;istio.&#x200b;diagnosticMetrics.&#x200b;enabled**  | boolean | If enabled, diagnostic metrics are collected. The default is `false`. |
| **input.&#x200b;istio.&#x200b;enabled**  | boolean | Enabled specifies if the 'istio' input is enabled, so that istio-proxy metrics are scraped from Pods that have the istio-proxy sidecar injected. The default is `false`. |
| **input.&#x200b;istio.&#x200b;envoyMetrics**  | object | EnvoyMetrics enables the collection of additional Envoy metrics with prefix `envoy_`. The default is `false`. |
| **input.&#x200b;istio.&#x200b;envoyMetrics.&#x200b;enabled**  | boolean | Enabled specifies that Envoy metrics with prefix `envoy_` are scraped additionally. The default is `false`. |
| **input.&#x200b;istio.&#x200b;namespaces**  | object | Namespaces configures the namespaces for which the collection should be activated. By default, all namespaces including system namespaces are enabled. |

> | **input.&#x200b;istio.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude telemetry data from the specified Namespace names only. By default all namespaces (except system namespaes dependent on input type). An exclude list cannot be specified together with an include list. |

Reviewer: ```suggestion
| **input.&#x200b;istio.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude telemetry data from the specified namespace names only. By default, all namespaces (depending on input type: except system namespaces) are included. You can't specify an exclude list together with an include list. |
```

> | **input.&#x200b;istio.&#x200b;namespaces.&#x200b;include**  | \[\]string | Include telemetry data from the specified Namespace names only. By default all namespaces (except system namespaes dependent on input type). An include list cannot be specified together with an exclude list. |

Reviewer: ```suggestion
| **input.&#x200b;istio.&#x200b;namespaces.&#x200b;include**  | \[\]string | Include telemetry data from the specified namespace names only. By default, all namespaces (depending on input type: except system namespaces) are included. You can't specify an include list together with an exclude list. |
```

| **input.&#x200b;otlp**  | object | OTLP input configures the push endpoint to receive metrics from an OTLP source. |

> | **input.&#x200b;otlp.&#x200b;disabled**  | boolean | Disabled specifies if the 'otlp' is deactivated, so that no push-based OTLP signals are collected. The default is `false`. |

Reviewer: ```suggestion
| **input.&#x200b;otlp.&#x200b;disabled**  | boolean | Disabled specifies if the 'otlp' input is deactivated, so that no push-based OTLP signals are collected. The default is `false`. |
```

| **input.&#x200b;prometheus.&#x200b;diagnosticMetrics**  | object | DiagnosticMetrics configures collection of additional diagnostic metrics. The default is `false`. |
| **input.&#x200b;prometheus.&#x200b;diagnosticMetrics.&#x200b;enabled**  | boolean | If enabled, diagnostic metrics are collected. The default is `false`. |
| **input.&#x200b;prometheus.&#x200b;enabled**  | boolean | Enabled specifies if the 'prometheus' input is enabled, so that whether Service endpoints and Pods marked with `prometheus.io/scrape=true` annotation are scraped. The default is `false`. |
| **input.&#x200b;prometheus.&#x200b;namespaces**  | object | Namespaces specifies from which namespaces metrics are collected. By default, all namespaces except the system namespaces are enabled. To enable all namespaces including system namespaces, use an empty struct notation. |
| **input.&#x200b;prometheus.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude telemetry data from the specified Namespace names only. By default all namespaces (except system namespaes dependent on input type). An exclude list cannot be specified together with an include list. |

> | **input.&#x200b;prometheus.&#x200b;namespaces.&#x200b;include**  | \[\]string | Include telemetry data from the specified Namespace names only. By default all namespaces (except system namespaes dependent on input type). An include list cannot be specified together with an exclude list. |

Reviewer: ```suggestion
| **input.&#x200b;prometheus.&#x200b;namespaces.&#x200b;include**  | \[\]string | Include telemetry data from the specified namespace names only. By default, all namespaces (depending on input type: except system namespaces). You can't specify an include list together with an exclude list. |
```

| **input.&#x200b;runtime**  | object | Runtime input configures collection of Kubernetes runtime metrics. |
| **input.&#x200b;runtime.&#x200b;enabled**  | boolean | Enabled specifies if the 'runtime' input is enabled, so that runtime metrics are collected. The default is `false`. |
| **input.&#x200b;runtime.&#x200b;namespaces**  | object | Namespaces specifies from which namespaces metrics are collected. By default, all namespaces except the system namespaces are enabled. To enable all namespaces including system namespaces, use an empty struct notation. |

> | **input.&#x200b;runtime.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude telemetry data from the specified Namespace names only. By default all namespaces (except system namespaes dependent on input type). An exclude list cannot be specified together with an include list. |

Reviewer: ```suggestion
| **input.&#x200b;runtime.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude telemetry data from the specified namespace names only. By default all namespaces (depending on input type: except system namespaces). You cannot specify an exclude list together with an include list. |
```
